### PR TITLE
Feature/audit missing drug class

### PR DESCRIPTION
**Pull request by pbugni**
In process of reporting missing drug classes:
- refactored session workarounds to `wrapped_session.py`
- added `audit.audit_entry()` to simplify auditing
- cleaned up imports, all now using `package.module` approach

### CODE REVIEW

File: sof_wrapper/wrapped_session.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        return get_redis_session_data(g.session_id).get(key, default)
+
+
+def set_session_value(key, value):
+    if 'session_id' not in g:
+        session[key] = value
+        return
+
+    session_data = get_redis_session_data(g.session_id)
+    session_data[key] = value
```
Comment (ivan-c) [inline sof_wrapper/wrapped_session.py:29]: Does this actually set `key` persistently? `get_redis_session_data()` doesn't return an actual session that will persist changes

File: sof_wrapper/audit.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+def audit_entry(message, level='info', extra=None):
+    """Log entry, adding in session info such as active user"""
+    try:
+        log_at_level = getattr(current_app.logger, level.lower())
+    except AttributeError:
+        raise ValueError(f"audit_entry given bogus level: {level}")
+
+    if extra is None:
+        extra = {}
+    extra['user'] = get_session_value('user')
```
Comment (ivan-c) [inline sof_wrapper/audit.py:18]: I think you need to check if user exists before adding it to `extra`?
```suggestion
    user = get_session_value('user')
    if user:
        extra['user'] = user
```

File: sof_wrapper/auth/views.py [hunk 3]
```diff
@@ -153,23 +154,20 @@ def authorize():
...
-        extra['user'] = {'username': username, 'DEA': DEA}
-
-    # retain extracted user details in session, as needed elsewhere
-    session['user'] = extra['user']
+        session['user'] = {'username': username, 'DEA': DEA}
 
     if 'patient' in token_response:
-        extra['subject'] = 'Patient/{}'.format(token_response['patient'])
-    current_app.logger.info("login", extra=extra)
+        session['subject'] = 'Patient/{}'.format(token_response['patient'])
```
Comment (ivan-c) [inline sof_wrapper/auth/views.py:168]: Should you also look for `launch_token_patient` ie
`get_session_value('token_response', {}).get('patient') or get_session_value('launch_token_patient')`?

File: sof_wrapper/wrapped_session.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
+import pickle
+from flask import g, current_app, session
+
+
+def get_session_value(key, default=None):
```
Comment (ivan-c) [inline sof_wrapper/wrapped_session.py:5]: Nice, the perfect util function!

**Review by ivan-c — COMMENTED** [review]
So far so good!

File: sof_wrapper/audit.py (hunk 2 above)
Comment (pbugni) [reply sof_wrapper/audit.py:18]: sure, i know it always will, but code changes, good suggestion.

File: sof_wrapper/wrapped_session.py (hunk 1 above)
Comment (pbugni) [reply sof_wrapper/wrapped_session.py]: updated to raise if we attempt to set a session variable w/o a session cookie.  i.e. catch the misleading no-op

File: sof_wrapper/auth/views.py (hunk 3 above)
Comment (pbugni) [reply sof_wrapper/auth/views.py]: we look there and capture in /launch - any reason it would be different here in /authorize ?

**Review by ivan-c — APPROVED** [review]
Looks good!

**Pull request merged**